sled-agent/src/sim/http_entrypoints.rs

```diff
@@ -741,6 +742,20 @@ impl SledAgentApi for SledAgentSimImpl {
     {
         method_unimplemented()
     }
+
+    async fn support_logs(
+        _request_context: RequestContext<Self::Context>,
+    ) -> Result<HttpResponseOk<SledDiagnosticsLogs>, HttpError> {
+        // We return an empty file index for support bundle lifecycle testing
```

Comment:
I think the other tests are enough to cover the logic of finding and validating log paths. If we want we could add something like a single log path to the index and for downloading the file we could send back a predefined set of bytes. Going a step further would mean we would need to mock more of what oxlog does which kind of feels wrong.